app/code/Magento/ConfigurableProduct/view/frontend/web/js/configurable.js

```diff
@@ -155,7 +160,13 @@ define([
 
                 if (element.value) {
                     attributeId = element.id.replace(/[a-z]*/, '');
-                    this.options.values[attributeId] = element.value;
+
+                    if (this.options.spConfig.attributes[attributeId] !== undefined
+                        && this.options.spConfig.attributes[attributeId].options.find(function(optionElement) {
```

Comment:
Same as above.

Reply:
yes you are correct, but i do apply the rule of three (https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) and therefor adapted the second occurrence as paste and copy (for my problem i wouldn't need that change, but added it for consistency)

Reply:
yes you are correct, but i do apply the rule of three (https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) and therefor adapted the second occurrence as paste and copy (for my problem i wouldn't need that change, but added it for consistency)

Reply:
We discuss this and still I would like to ask you to make this change. Such comparisons are part of the legacy code and we are trying to refactor them as we go.